apps/prairielearn/src/ee/lib/billing/components/InstructorInstanceAdminBillingForm.html.ts

```diff
@@ -0,0 +1,247 @@
+  const computeEnabledByInstitution = planGrantsMatchPlanFeatures(
+    input.institutionPlanGrants,
+    'compute',
+  );
+  const computeEnabledByCourseInstance = planGrantsMatchPlanFeatures(
+    input.courseInstancePlanGrants,
+    'compute',
+  );
+  const computeEnabled =
+    (!studentBillingEnabled && (computeEnabledByInstitution || computeEnabledByCourseInstance)) ||
+    input.desiredRequiredPlans.includes('compute');
+
+  let studentBillingCanChange = true;
+  const studentBillingDidChange =
+    input.initialRequiredPlans.includes('basic') !== input.desiredRequiredPlans.includes('basic');
+  let studentBillingAlert: AlertProps | null = null;
+  if (studentBillingInitialEnabled && input.enrollmentCount > input.enrollmentLimit) {
+    studentBillingCanChange = false;
+    const inflectedCountVerb = input.enrollmentCount === 1 ? 'is' : 'are';
+    const inflectedCountNoun = input.enrollmentCount === 1 ? 'enrollment' : 'enrollments';
+    studentBillingAlert = {
+      message: [
+        `There ${inflectedCountVerb} ${input.enrollmentCount} ${inflectedCountNoun} in this course, which exceeds the limit of ${input.enrollmentLimit}.`,
+        'To disable student billing, first remove excess enrollments.',
+      ].join(' '),
+      color: 'warning',
+    };
+  }
+
+  let computeCanChange = true;
+  const computeDidChange =
+    input.initialRequiredPlans.includes('compute') !==
+    input.desiredRequiredPlans.includes('compute');
+  let computeAlert: AlertProps | null = null;
+  if (!studentBillingEnabled && (computeEnabledByInstitution || computeEnabledByCourseInstance)) {
+    computeCanChange = false;
+    computeAlert = {
+      message: computeEnabledByCourseInstance
+        ? 'This course instance already has access to compute features without additional payment.'
+        : 'Courses in your institution already have access to compute features without additional payment.',
+      color: 'success',
+    };
+  }
+
+  return {
+    studentBillingEnabled,
+    studentBillingCanChange,
+    studentBillingDidChange,
+    studentBillingAlert,
+    computeEnabled,
+    computeCanChange,
+    computeDidChange,
+    computeAlert,
+  };
+}
+
+export function InstructorInstanceAdminBillingForm(
+  props: InstructorInstanceAdminBillingInput & {
+    enrollmentLimitSource: 'course_instance' | 'institution';
+    externalGradingQuestionCount: number;
+    workspaceQuestionCount: number;
+    csrfToken: string;
+  },
+) {
+  const {
+    enrollmentCount,
+    enrollmentLimit,
+    enrollmentLimitSource,
+    externalGradingQuestionCount,
+    workspaceQuestionCount,
+    csrfToken,
+  } = props;
+
+  const {
+    studentBillingEnabled,
+    studentBillingCanChange,
+    studentBillingAlert,
+    computeEnabled,
+    computeCanChange,
+    computeAlert,
+  } = instructorInstanceAdminBillingState(props);
+
+  const enrollmentLimitText = studentBillingEnabled ? '∞' : enrollmentLimit;
```

Comment:
I personally found the infinity sign to be a bit overly cute. I'd prefer to just write "XX enrollments" without the `/ ∞` in this case (and keep the help text that explains we have unlimited enrollments because student billing is enabled). I'd also prefer to not show the (empty) progress bar when we have unlimited enrollments.

Reply:
Hah, sure, I can remove the infinity sign. I'm not sure what to do about the progress bar though - showing/hiding it would cause layout shift that would move the checkbox out from under the cursor, which is what I was trying to avoid by having it become empty.

Reply:
I agree having layout jump around is annoying. Two ideas:
1. Use `visibility: hidden` to hide the progress bar but leave the space intact.
2. Shift the progress bar to be displayed on the same line as the count, so we'd have:
    ```
    12 / 20 enrollments [xxxxxxxx---------]
    ```
    This would require using a fixed width progress bar, which I think would look better anyway (the current progress bar is so long that I initially thought it was a thick horizontal line rather than a progress bar.

Without having tried them I think I like option (2).

Reply:
I switched to a width-capped, inline progress bar in 277f732. I fudged the numbers a bit and enforced a minimum width of 2% so that it wouldn't disappear completely for low enrollment counts with a high enrollment limit.

<img width="547" alt="Screenshot 2023-07-21 at 13 02 34" src="https://github.com/PrairieLearn/PrairieLearn/assets/1476544/5c6cdd4b-6f6c-4ab7-8975-0fb71dfc9042">